test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/RegressionJSTest.scala

```diff
@@ -96,6 +99,40 @@ class RegressionJSTest {
     assertEquals(6, obj2.bar)
   }
 
+  @Test def preserveSideEffectsOfJSOps_Issue4621(): Unit = {
+    if (jsBigInts) {
+      @noinline def bi(x: Int): js.BigInt = js.BigInt(x)
+
+      // These must be stored as `val`s first
```

Comment:
Consider adding " to trigger the original problem". (Arguably that duplicates context that exists, but IMO it is a bit hard to grok otherwise).